repos/fdbt-site/src/interfaces/typeGuards.ts

```diff
@@ -163,9 +166,18 @@ export const isPeriodExpiry = (
     periodExpiryAttribute: PeriodExpiry | ErrorInfo[] | undefined,
 ): periodExpiryAttribute is PeriodExpiry => !!periodExpiryAttribute && 'productValidity' in periodExpiryAttribute;
 
+export const isDayOfTheWeek = (dayOfTheWeek: DayOfTheWeek | ErrorInfo[] | undefined): dayOfTheWeek is DayOfTheWeek =>
+    !!dayOfTheWeek;
+
+export const isCapStart = (capStart: CapStart | ErrorInfo[] | undefined): capStart is CapStart => !!capStart;
+
 export const isCapExpiry = (capExpiryAttribute: CapExpiry | ErrorInfo[] | undefined): capExpiryAttribute is CapExpiry =>
     !!capExpiryAttribute && 'productValidity' in capExpiryAttribute;
 
+export const isCapStartInfo = (
+    capStartAttribute: CapStartInfo | ErrorInfo[] | undefined,
+): capStartAttribute is CapStartInfo => !!capStartAttribute && 'type' && 'start' in capStartAttribute;
```

Comment:
'type' here is being judged to be true or false, because it is between two & symbols. If you want to check if 'type' is in capStartAttribute and 'start' is in capStartAttribute, you have to write them both out.

Reply:
!!capStartAttribute && 'type' && 'start' in capStartAttribute;

true/false && true && true/false